Content.Shared/Alert/AlertsSystem.cs

```diff
@@ -71,107 +124,153 @@ public bool TryGetAlertState(EntityUid euid, AlertKey key, out AlertState alertS
+        if (!_alertsQuery.Resolve(entity, ref entity.Comp, false))
             return;
 
-        if (TryGet(alertType, out var alert))
+        if (!TryGet(alertType, out var alert))
         {
-            // Check whether the alert category we want to show is already being displayed, with the same type,
-            // severity, and cooldown.
-            if (alertsComponent.Alerts.TryGetValue(alert.AlertKey, out var alertStateCallback) &&
-                alertStateCallback.Type == alertType &&
-                alertStateCallback.Severity == severity &&
-                alertStateCallback.Cooldown == cooldown &&
-                alertStateCallback.AutoRemove == autoRemove &&
-                alertStateCallback.ShowCooldown == showCooldown)
-            {
-                return;
-            }
+            Log.Error($"Unable to show alert {alertType}, please ensure this alertType has a corresponding YML alert prototype");
+            return;
+        }
 
-            // In the case we're changing the alert type but not the category, we need to remove it first.
-            alertsComponent.Alerts.Remove(alert.AlertKey);
+        // Check whether the alert category we want to show is already being displayed, with the same type,
+        // severity, and cooldown.
+        if (entity.Comp.Alerts.TryGetValue(alert.AlertKey, out var alertStateCallback) &&
+            alertStateCallback.Type == alertType &&
+            alertStateCallback.Severity == severity &&
+            alertStateCallback.Cooldown == cooldown &&
+            alertStateCallback.AutoRemove == autoRemove &&
+            alertStateCallback.ShowCooldown == showCooldown)
+        {
+            return;
+        }
 
-            var state = new AlertState
-                { Cooldown = cooldown, Severity = severity, Type = alertType, AutoRemove = autoRemove, ShowCooldown = showCooldown};
-            alertsComponent.Alerts[alert.AlertKey] = state;
+        // In the case we're changing the alert type but not the category, we need to remove it first.
+        entity.Comp.Alerts.Remove(alert.AlertKey);
 
-            // Keeping a list of AutoRemove alerts, so Update() doesn't need to check every alert
-            if (autoRemove)
-            {
-                var autoComp = EnsureComp<AlertAutoRemoveComponent>(euid);
-                if (!autoComp.AlertKeys.Contains(alert.AlertKey))
-                    autoComp.AlertKeys.Add(alert.AlertKey);
-            }
-
-            AfterShowAlert((euid, alertsComponent));
+        var state = new AlertState
+            { Cooldown = cooldown, Severity = severity, Type = alertType, AutoRemove = autoRemove, ShowCooldown = showCooldown};
+        entity.Comp.Alerts[alert.AlertKey] = state;
 
-            Dirty(euid, alertsComponent);
+        // Keeping a list of AutoRemove alerts, so Update() doesn't need to check every alert
+        if (autoRemove)
+        {
+            var autoComp = EnsureComp<AlertAutoRemoveComponent>(entity);
```

Comment:
Man I wish alerts were just entities so we could just TimedDespawn them or something.